modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/GridCacheQueryAdapter.java

```diff
@@ -625,7 +629,7 @@ private <R> CacheQueryFuture<R> execute0(@Nullable IgniteReducer<T, R> rmtReduce
         GridCloseableIterator it;
 
         if (loc)
-            it = qryMgr.scanQueryLocal(this, true);
+            it = qryMgr.queryLocal(this, true);
```

Comment:
It looks strange for non-local query we call `scanQueryDistributed` while for local it calls generic method. Let's separate `scanQueryLocal` and `indexQueryLocal`.

Reply:
Done.